nf_core/lint/schema_lint.py

```diff
@@ -69,10 +85,16 @@ def schema_lint(self):
     # Lint the schema
     self.schema_obj = nf_core.schema.PipelineSchema()
     self.schema_obj.get_schema_path(self.wf_path)
+
     try:
-        self.schema_obj.load_lint_schema()
+        self.schema_obj.load_schema()
+        self.schema_obj.get_schema_defaults()
+        self.schema_obj.validate_schema()
+        # Validate default parameters, ignoring required ones as they might be empty
+        schema_no_required = remove_required_fields(self.schema_obj.schema)
+        jsonschema.validate(self.schema_obj.schema_defaults, schema_no_required)
         passed.append("Schema lint passed")
-    except AssertionError as e:
+    except Exception as e:
```

Comment:
Don't love this - generally a good idea to never catch the base `Exception` class as other stuff can slip by unnoticed (speaking from experience! 😓 )

Better to just catch both exception types, eg:

```python
    except (AssertionError, TypeError) as e:
```

Reply:
Thanks!

Reply:
The classic unexpected effect would be that that this would catch the user pressing `ctrl-c` for example